src/Styleguide/Elements/Responsive.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react"
+
+const ResponsiveContext = React.createContext({})
+
+interface Breakpoints {
+  [breakpoint: string]: string
+}
+
+interface ResponsiveProviderProps {
+  initialBreakpoint?: string
+  breakpoints: Breakpoints
+}
+
+interface BreakpointState {
+  [breakpoint: string]: boolean
+}
+
+interface ResponsiveProviderState {
+  breakpoints: BreakpointState
+  mediaMatchers: MediaQueryList[]
+  mqHandler: MediaQueryListListener | null
+}
+
+export class ResponsiveProvider extends React.Component<
+  ResponsiveProviderProps,
+  ResponsiveProviderState
+> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      breakpoints: {
+        ...Object.keys(props.breakpoints)
+          .map(breakpoint => ({
+            [breakpoint]: breakpoint === props.initialBreakpoint ? true : false,
+          }))
+          .reduce((acc, curr) => ({ ...acc, ...curr }), {}),
+      },
+      mediaMatchers: [],
+      mqHandler: null,
+    }
+  }
+
+  componentDidMount() {
+    this.setupObservers(this.props.breakpoints)
+  }
+
+  componentWillUnmount() {
+    if (this.state.mqHandler) {
+      this.state.mediaMatchers.forEach(mediaQuery =>
+        mediaQuery.removeListener(this.state.mqHandler)
+      )
+    }
+  }
+
+  setupObservers = breakpoints => {
+    const breakpointKeys = Object.keys(breakpoints)
+    const mediaMatchers = breakpointKeys.map(breakpoint =>
+      window.matchMedia(breakpoints[breakpoint])
+    )
+    const mqHandler = () => {
+      for (let i = 0; i < mediaMatchers.length; ++i) {
+        this.setState({
+          breakpoints: { [breakpointKeys[i]]: mediaMatchers[i].matches },
+        })
+      }
+    }
+    this.setState({ mediaMatchers, mqHandler })
+    mediaMatchers.forEach(mediaQuery => {
+      mediaQuery.addListener(mqHandler)
+    })
+    mqHandler()
+  }
+
+  render() {
+    return (
+      <ResponsiveContext.Provider value={this.state.breakpoints}>
```

Comment:
We'll need to upgrade force to `16.3` but that shouldn't be a problem; there are no breaking changes, just a deprecated context API that's still available for use. This kind of context tends to be locally scoped and deliberate, so no real conflict worry anyway.

Reply:
Actually @zephraph, I take that back. Lets use [this](https://github.com/jamiebuilds/create-react-context) for now and then once Force is upgraded we can use native API.

Reply:
Done. I'll create a PR both on reaction and force to upgrade `React`

Reply:
word 👍